src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1090,6 +1120,126 @@ public void UpdateBaseline(long timestampDiff)
+                        {
+                            currentLocalTime = originalBaseline.AddMilliseconds(localTimestamp);
+                            if (isAppend)
+                            {
+                                splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                continue;
+                            }
+
+                            var compareTags = localTag.ContentAsSpan().SequenceCompareTo(tagSlice);
+                            if (compareTags < 0) // local tag wins
+                            {
+                                splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                continue;
+                            }
+
+                            if (compareTags == 0) // equal tags
+                            {
+                                if (FromReplication)
+                                {
+                                    var compareValues = localValues.SequenceCompareTo(newValues);
+                                    var isIncrement = tagSlice.StartsWith(TimedCounterPositivePrefixBuffer);
+
+                                    if (isIncrement) // increment entry --> take higher values
+                                    {
+                                        if (compareValues >= 0)
+                                            splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                        else
+                                            splitSegment.Append(_context.Allocator, 0, newValues, tagSlice, status);
+                                    }
+                                    else // decrement entry --> take lower values
+                                    {
+                                        if (compareValues <= 0)
+                                            splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                        else
+                                            splitSegment.Append(_context.Allocator, 0, newValues, tagSlice, status);
+                                    }
+
+                                    isAppend = true;
+                                    continue;
+                                }
+
+                                int length = Math.Min(localValues.Length, newValues.Length);
+                                for (int i = 0; i < length; i++)
+                                {
+                                    newValues[i] += localValues[i];
+                                }
+                            }
+
+                            isAppend = true;
+                            splitSegment.Append(_context.Allocator, 0, newValues, tagSlice, status);
+                        }
+                    }
+                }
```

Comment:
Do I miss something here? Where do you actually persist those newly generated segment?

Reply:
We need to split not from the current spot. In the original case, we flush the segment, initialize it and try to append the entry again (to the initialized segment). In this case, we must find the first entry with the timestamp and flush before the appearance of this entry. Then, we need to append the new entry in the correct offset according to the tag. 
We continue working on an initialized segment - no need to append again (as the original case behavior).